core/arch/arm/include/kernel/generic_boot.h

```diff
@@ -31,10 +31,12 @@
 
 #if defined(CFG_WITH_ARM_TRUSTED_FW)
 uint32_t cpu_on_handler(uint32_t a0, uint32_t a1);
-uint32_t *generic_boot_init_primary(uint32_t pageable_part);
+uint32_t *generic_boot_init_primary(uint32_t pageable_part, uint32_t unused,
+				    uint32_t fdt);
```

Comment:
Are you saying that `paddr_t` is not always suitable to represent a physical address? This would be confusing.
How can `paddr_t` be 64-bit on ILP32? It is defined as `uintptr_t` which is an integer the size of a pointer, so if platform is ILP32 this is 32 bits.
The problem I see here is that `fdt` (and `pageable_part` and `nsec_entry` below, for that matter) cannot have a high physical address. Why should we assume any specific physical location for those?
Am I missing something?